Update Kotlin to `2.2.0`

oguzkocer opened the pull request:
## Summary
- Fixed deprecated string case conversion methods
- Added annotation default target compiler flag
- Resolved various Kotlin compiler warnings

## Changes
- Replace `toLowerCase(Locale)` with `lowercase(Locale)` 
- Replace `toUpperCase(Locale)` with `uppercase(Locale)`
- Add `-Xannotation-default-target=param-property` compiler flag
- Fix redundant else in exhaustive when statement
- Fix reified type intersection warning with explicit type parameter
- Align parameter names with interface definitions

oguzkocer committed:
Update `kotlin-main` to `2.2.0` & `ksp` to `2.2.0-2.0.2`

oguzkocer committed:
Replace deprecated toLowerCase/toUpperCase with lowercase/uppercase

Fixed compilation warnings by updating deprecated string case conversion methods
to use the new Kotlin standard library functions while preserving locale behavior.

Changes:
- Replace toLowerCase(Locale.ROOT) with lowercase(Locale.ROOT)
- Replace toLowerCase(Locale.getDefault()) with lowercase(Locale.getDefault())
- Replace toUpperCase(Locale.US) with uppercase(Locale.US)

oguzkocer committed:
Add annotation default target compiler flag

Resolves Kotlin 2.0+ warnings about annotation targets by opting into the future
behavior where annotations on constructor parameters apply to both the parameter
and generated property.

Changes:
- Add -Xannotation-default-target=param-property to Kotlin compiler args
- Eliminates ~2000 annotation target warnings across the codebase

oguzkocer committed:
Remove redundant else from exhaustive when statement

Fixed Kotlin compiler warning by removing unnecessary else branch from
exhaustive when statement covering all FeatureFlagValueSource enum values.

Changes:
- Remove redundant else -> null branch from toUiValue() function
- Change return type from String? to String since null is no longer possible

oguzkocer committed:
Fix parameter name mismatch in EditPostActivity

Align parameter name with interface definition to avoid Kotlin compiler
warning about named argument usage.

Changes:
- Rename onExceptionSend to onSendJsException to match JsExceptionListener interface

oguzkocer committed:
Fix parameter name mismatch in StatsFragment

Align parameter name with interface definition to avoid Kotlin compiler
warning about named argument usage.

Changes:
- Rename pullToRefreshEmitterListener to emitterListener to match PullToRefreshReceiverListener interface

oguzkocer committed:
Fix reified type intersection warning in MySiteFragment

Added explicit java.io.Serializable type parameter to getSerializableCompat
to avoid Kotlin compiler inferring intersection type and to resolve ambiguity
between kotlin.io.Serializable and java.io.Serializable.

Changes:
- Use explicit java.io.Serializable type parameter for getSerializableCompat
- Maintain safe cast to QuickStartStore.QuickStartTask since interface doesn't implement Serializable

oguzkocer committed:
Fix callback parameter name mismatch in EditPostActivity

Align anonymous callback parameter name with external JsExceptionCallback
interface to avoid Kotlin compiler warning about named argument usage.

Changes:
- Rename success parameter to sent in onReportSent callback implementation

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba commented:
Let's ignore these tests, then. 

Generally speaking, since some time I wanted to propose removing `processors` module as to me this has much higher costs of maintenance comparing to the value for the team: [AINFRA-46: Remove internal annotation processor from WordPress/Jetpack](https://linear.app/a8c/issue/AINFRA-46/remove-internal-annotation-processor-from-wordpressjetpack).

oguzkocer committed:
Ignore RemoteConfigProcessorTest tests for Kotlin 2.2.0 compatibility

Added @Ignore annotations to all test methods in RemoteConfigProcessorTest
to prevent compilation issues with Kotlin 2.2.0. These tests rely on
compiler testing utilities that are not compatible with the newer Kotlin version.

Changes:
- Added @Ignore import
- Applied @Ignore annotation to all 4 test methods with descriptive reason
- Tests now skip during execution preventing build failures

nbradbury reviewed (approved):
Tested on multiple Android versions and all good! :shipit:
